app/api/chemotion/reaction_api.rb

```diff
@@ -78,7 +82,7 @@ class ReactionAPI < Grape::API
           Reaction.joins(:collections).where('collections.user_id = ?', current_user.id).uniq
         end.order("created_at DESC")
 
-        paginate(scope).map{|s| ElementPermissionProxy.new(current_user, s).serialized}
+        paginate(scope).map{|s| ElementPermissionProxy.new(@current_user, s, @user_ids).serialized}
```

Comment:
We can use ```current_user```, since the method is in app/api/api.rb.
We should also create ```user_ids``` methods.